app/services/analytics.rb

```diff
@@ -11,6 +11,7 @@ def track_event(event, attributes = {})
     analytics_hash = {
       event_properties: attributes.except(:user_id),
       new_session_path: @first_path_visit_this_session,
+      path: request&.respond_to?(:path) ? request.path : '',
```

Comment:
when (outside of FakeRequest) does `request` not have a path? FakeRequest is our class, we can just update it, right?

Also, what does having an empty string `''` get us vs just having it be nil?

```suggestion
      path: request&.path,
```

Reply:
Sometimes the request is nil for background job, etc.

Reply:
Right, but then the `&` handles the nil part, no need to add `respond_to?` and then ... `nil.respond_to?` is also safe so basically I'm asking why both `.&` and `respond_to?`

Reply:
ah.  i just added path to FakeRequest so I can take that out now